src/main/java/org/spongepowered/common/mixin/core/server/management/MixinPlayerInteractionManager.java

```diff
@@ -326,6 +258,117 @@ public EnumActionResult processRightClickBlock(EntityPlayer player, net.minecraf
         // } // Sponge - Remove unecessary else bracket
     }
 
+    /**
+     * @reason Due to the way that buckets and the like can be handled
+     * on the client, often times we need to cancel the item stack usage
+     * due to server side cancellation logic that may not exist on the client.
+     * Therefor, the cancellation of possible block changes doesn't take
+     * effect, and therefor requires telling the client to set back the item
+     * in hand.
+     * @reason Fire interact item event.
+     */
+    @Overwrite
+    public EnumActionResult processRightClick(EntityPlayer player, net.minecraft.world.World worldIn, ItemStack stack, EnumHand hand) {
+        if (this.gameType == GameType.SPECTATOR) {
+            return EnumActionResult.PASS;
+        }
+
```

Comment:
Can you comment where forte patches to this method occur, so we don’t have o look out for that on the side, but in the overwrite itself?

Reply:
They don't. 
Forge only adds its event and check for cancellation 
https://github.com/MinecraftForge/MinecraftForge/blob/a5f29570a39508afe96870e7e665ff4bbf964aa8/patches/minecraft/net/minecraft/server/management/PlayerInteractionManager.java.patch#L199